cachito/web/models.py

```diff
@@ -743,3 +760,80 @@ def to_json(self):
         :rtype: str
         """
         return self.name
+
+
+class ConfigFileBase:
+    """A base class with attributes common to all configuration file classes."""
+
+    id = db.Column(db.Integer, primary_key=True)
+    # This is the relative path of where the file should be in the extracted bundle
+    path = db.Column(db.String, nullable=False, index=True)
+
+
+class ConfigFileBase64(ConfigFileBase, db.Model):
+    """A configuration file that the consumer must set for the bundle to be usable."""
+
+    __table_args__ = (db.UniqueConstraint("content", "path"),)
+    content = db.Column(db.String, nullable=False, index=True)
+
+    @classmethod
+    def get_or_create(cls, path, content):
+        """
+        Get the configuration file from the database and create it if it doesn't exist.
+
+        :param str path: the relative path of where the file should be in the bundle
+        :param str content: the base64 string of the content
+        :return: a ConfigFileBase64 object based on the input; the ConfigFileBase64 object will be
+            added to the database session, but not committed if it was created
+        :rtype: ConfigFileBase64
+        """
+        config_file = cls.query.filter_by(path=path, content=content).first()
+        if not config_file:
+            config_file = cls(path=path, content=content)
+            db.session.add(config_file)
+
+        return config_file
+
+    def to_json(self):
+        """
+        Generate the JSON representation of the configuration file.
+
+        :return: the JSON representation of the configuration file.
+        :rtype: dict
+        """
+        return {"content": self.content, "path": self.path, "type": "base64"}
+
+    @classmethod
+    def validate_json(cls, payload):
+        """
+        Validate the input configuration file.
+
+        :param list payload: the list of dictionaries of configuration files
+        :raises ValidationError: if the configuration file is invalid
+        """
+        if not isinstance(payload, dict):
+            raise ValidationError("The base64 configuration file must be a JSON object")
```

Comment:
If type is a parameter, it doesn't right to hard code `base64` in these log messages.

Reply:
@lcarva after thinking about this, I'll move most of this method to the base class and require each child class have the `type_name` class property.

Reply:
IMO base class is unnecessary when there's only a single type being introduced, but I'll leave that up to you :)